kmk/scanners/digitalio.py

```diff
@@ -125,15 +130,19 @@ def scan_for_changes(self):
                         row = oidx
                         col = iidx
 
-                    pressed = new_val
+                    pressed = (
+                        not new_val
+                        if self.resistor_pull is digitalio.Pull.UP
+                        else new_val
+                    )
```

Comment:
```suggestion
                    if self.resistor_pull is digitalio.Pull.UP:
                        pressed = not new_val
                    else:
                        pressed = new_val
```
An inline-if is harder to read and doesn't safe LOC compared to a regular if-statement.
alternatively:
```
pressed = new_val
if self.resistor_pull is digitalio.Pull.UP:
    pressed = not pressed
```

Also, I think it should be `if pull == digitalio.Pull.UP`, since semantically we're comparing values, not object identities.

Reply:
Good call on readability. Will incorporate that change.

For enum comparison, see this documentation:
>Enumeration members are compared by identity:
>```
>>>> Color.RED is Color.RED
>True
>```